federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -142,24 +171,69 @@ func NewIngressController(client federation_release_1_4.Interface) *IngressContr
 			ClusterAvailable: func(cluster *federation_api.Cluster) {
-				// When new cluster becomes available process all the ingresses again.
-				ic.clusterDeliverer.DeliverAt(allClustersKey, nil, time.Now().Add(ic.clusterAvailableDelay))
+				// When new cluster becomes available process all the ingresses again, and configure it's ingress controller's configmap with the correct UID
+				ic.clusterDeliverer.DeliverAfter(cluster.Name, cluster, ic.clusterAvailableDelay)
 			},
 		},
 	)
 
-	// Federated updater along with Create/Update/Delete operations.
-	ic.federatedUpdater = util.NewFederatedUpdater(ic.ingressFederatedInformer,
+	// Federated informer on configmaps for ingress controllers in members of the federation.
+	ic.configMapFederatedInformer = util.NewFederatedInformer(
+		client,
+		func(cluster *federation_api.Cluster, targetClient kube_release_1_4.Interface) (cache.Store, framework.ControllerInterface) {
+			glog.V(4).Infof("Returning new informer for cluster %q", cluster.Name)
+			return framework.NewInformer(
+				&cache.ListWatch{
+					ListFunc: func(options api.ListOptions) (pkg_runtime.Object, error) {
+						if targetClient == nil {
+							glog.Errorf("targetClient is nil")
+						}
+						return targetClient.Core().ConfigMaps(uidConfigMapNamespace).List(options) // we only want to list one by name - unfortunately Kubernetes don't have a selector for that.
+					},
+					WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+						return targetClient.Core().ConfigMaps(uidConfigMapNamespace).Watch(options) // as above
+					},
+				},
+				&v1.ConfigMap{},
+				controller.NoResyncPeriodFunc(),
+				// Trigger reconcilation whenever the ingress controller's configmap in a federated cluster is changed. In most cases it
+				// would be just confirmation that the configmap for the ingress controller is correct.
+				util.NewTriggerOnAllChanges(
+					func(obj pkg_runtime.Object) {
+						ic.deliverConfigMapObj(cluster.Name, obj, ic.configMapReviewDelay, false)
+					},
+				))
+		},
+
+		&util.ClusterLifecycleHandlerFuncs{
+			ClusterAvailable: func(cluster *federation_api.Cluster) {
+				ic.clusterDeliverer.DeliverAfter(cluster.Name, cluster, ic.clusterAvailableDelay)
+			},
+		},
+	)
+
+	// Federated ingress updater along with Create/Update/Delete operations.
+	ic.federatedIngressUpdater = util.NewFederatedUpdater(ic.ingressFederatedInformer,
 		func(client kube_release_1_4.Interface, obj pkg_runtime.Object) error {
 			ingress := obj.(*extensions_v1beta1.Ingress)
 			glog.V(4).Infof("Attempting to create Ingress: %v", ingress)
 			_, err := client.Extensions().Ingresses(ingress.Namespace).Create(ingress)
+			if err != nil {
+				glog.Errorf("Error creating ingress %q: %v", types.NamespacedName{Name: ingress.Name, Namespace: ingress.Namespace}, err)
+			} else {
+				glog.V(4).Infof("Successfully created ingress %q", types.NamespacedName{Name: ingress.Name, Namespace: ingress.Namespace})
```

Comment:
Agreed.  Events need to be added throughout.  I'll do all of that in a separate PR.